### PR TITLE
Fix EZP-24760: console commands should validate siteaccess

### PR DESCRIPTION
**Pull request by wizhippo**

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/Console/Application.php [hunk 1]
```diff
@@ -28,4 +31,20 @@ public function __construct(KernelInterface $kernel)
...
     }
+
+    public function doRun(InputInterface $input, OutputInterface $output)
+    {
+        $kernel = $this->getKernel();
+        $kernel->boot();
+
+        $container = $kernel->getContainer();
+
+        $siteAccess = $input->getParameterOption('--siteaccess', null);
```
Comment (wizhippo) [inline eZ/Bundle/EzPublishCoreBundle/Console/Application.php]: should fall back to default

**Comment by andrerom** [comment]
+1, then my PR can be closed i think :)

**Comment by wizhippo** [comment]
@andrerom Yes, sorry thought I wouldn't have time tonight.

**Comment by andrerom** [comment]
No worries, I did not communicate I would jump on it either :)

**Comment by andrerom** [comment]
+1, ping @lolautruche

File: eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php [hunk 2]
```diff
@@ -42,9 +48,10 @@ class ConsoleCommandListener implements EventSubscriberInterface, SiteAccessAwar
     /**
      * ConsoleCommandListener constructor.
      */
-    public function __construct($defaultSiteAccessName, EventDispatcherInterface $eventDispatcher)
+    public function __construct($defaultSiteAccess, $siteAccessList, EventDispatcherInterface $eventDispatcher)
```
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php]: Why did you change `$defaultSiteAccessName` to `$defaultSiteAccess`? Usually SiteAccess is an object, so this makes it confusing here (same for property name).
Besides, `$siteAccessList` should be  type hinted as array :wink:

File: eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        $this->dispatcher->expects($this->once())
+            ->method('dispatch');
+        $input = new ArrayInput(array(), $this->inputDefinition);
+        $event = new ConsoleCommandEvent($this->command, $input, $this->testOutput);
+        $this->listener->onConsoleCommand($event);
+        $this->assertEquals(new SiteAccess('default', 'cli'), $this->siteAccess);
+    }
+}
+
+class TestOutput extends Output
```
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php]: CS: Only one class per file please (PSR-1).

**Comment by lolautruche** [comment]
+1 besides nitpicks

File: eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php (hunk 2 above)
Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php]: I guess he reflect the change in config name used to fix the issue regarding default site access on commands.

File: eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php (hunk 3 above)
Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ConsoleCommandListenerTest.php]: We have some exceptions to that rule for tests where we have some code outside of main class, but that is usually smaller so I agree it should be separate file here.

File: eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php (hunk 2 above)
Comment (wizhippo) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/ConsoleCommandListener.php]: @andrerom is correct.  I guess `%ezpublish.siteaccess.default%` is just a name so I will revert back.
